src/core_plugins/kibana/public/visualize/embeddable/visualize_embeddable_factory.js

```diff
@@ -92,10 +93,69 @@ export class VisualizeEmbeddableFactory extends EmbeddableFactory {
           handler.destroy();
         });
 
+        function hasQueryOrFilters(searchSource) {
+          const filters = searchSource.get('filter');
+          if (filters.length > 0) {
+            return true;
+          }
+
+          const query = searchSource.get('query');
+          if (query.query && query.query.length > 0) {
+            return true;
+          }
+
+          return false;
+        }
+
+        function getFilterLabel(filter) {
+          if (!_.has(filter, 'meta') || (!filter.meta.alias &&
```

Comment:
It seems like this isn't the only place we might be summarizing filters like this. Is there an existing function we can use? If not, this is just fine. If there are similar functions laying around, it seems like it belongs in a `Filter` module or something.